src/main/java/dan200/computercraft/core/terminal/TextBuffer.java

```diff
@@ -61,76 +27,34 @@ public int length()
 
     public String read()
     {
-        return read( 0, text.length );
-    }
-
-    public String read( int start )
-    {
-        return read( start, text.length );
-    }
-
-    public String read( int start, int end )
-    {
-        start = Math.max( start, 0 );
-        end = Math.min( end, text.length );
-        int textLength = Math.max( end - start, 0 );
-        return new String( text, start, textLength );
+        return this.toString();
     }
 
     public void write( String text )
     {
-        write( text, 0, text.length() );
+        write( text, 0 );
     }
 
     public void write( String text, int start )
     {
-        write( text, start, start + text.length() );
-    }
-
-    public void write( String text, int start, int end )
-    {
-        int pos = start;
         start = Math.max( start, 0 );
-        end = Math.min( end, pos + text.length() );
-        end = Math.min( end, this.text.length );
+        int end = Math.min( start + text.length(), this.text.length );
         for( int i = start; i < end; i++ )
         {
-            this.text[i] = text.charAt( i - pos );
+            this.text[i] = text.charAt( i - start );
         }
     }
 
     public void write( TextBuffer text )
     {
-        write( text, 0, text.length() );
-    }
-
-    public void write( TextBuffer text, int start )
-    {
-        write( text, start, start + text.length() );
-    }
-
-    public void write( TextBuffer text, int start, int end )
-    {
-        int pos = start;
-        start = Math.max( start, 0 );
-        end = Math.min( end, pos + text.length() );
-        end = Math.min( end, this.text.length );
-        for( int i = start; i < end; i++ )
-        {
-            this.text[i] = text.charAt( i - pos );
-        }
+        this.write( text.toString() );
```

Comment:
I think it's probably worth special-casing this one and doing it as an array copy. Just so we can avoid allocating an extra array.

Reply:
Updated to be similar to write( String text, int start )